### PR TITLE
Mfeigl/mobilemap unpack workflow

### PR DESCRIPTION
**Pull request by mfeigl**
Open Mobile Map Package checks MMPK files whether they need to be unpacked before they can be loaded; and if so, unpacks them before loading. Some MMPK files contain data which need to be unpacked before they could be read and used.

### CODE REVIEW

File: ArcGISRuntimeSDKQt_CppSamples/Maps/OpenMobileMap_MapPackage/OpenMobileMap_MapPackage.cpp [hunk 1]
```diff
@@ -16,16 +16,89 @@
...
+
+} // namespace
+
 OpenMobileMap_MapPackage::OpenMobileMap_MapPackage(QQuickItem* parent) :
-    QQuickItem(parent)
+  QQuickItem(parent)
 {
+  // create the MMPK data path
+  // data is downloaded automatically by the sample viewer app. Instructions to download
+  // seperately are specified in the readme.
```
Comment (lsmallwood) [inline ArcGISRuntimeSDKQt_CppSamples/Maps/OpenMobileMap_MapPackage/OpenMobileMap_MapPackage.cpp]: typo "seperately" => "separately" (I think...)

File: ArcGISRuntimeSDKQt_CppSamples/Maps/OpenMobileMap_MapPackage/OpenMobileMap_MapPackage.cpp [hunk 2]
```diff
@@ -38,30 +111,67 @@ void OpenMobileMap_MapPackage::init()
...
-    // load the mobile map package
-    m_mobileMapPackage->load();
-    //! [open mobile map package cpp snippet]
+      // set the map view's map to the first map in the mobile map package
+      setMapToView();
+    }
+  });
+
+  connect(m_mobileMapPackage, &MobileMapPackage::doneLoading, this, &OpenMobileMap_MapPackage::packageLoaded);
+
```
Comment (lsmallwood) [inline ArcGISRuntimeSDKQt_CppSamples/Maps/OpenMobileMap_MapPackage/OpenMobileMap_MapPackage.cpp]: I'd move the empty line after the closing tag of the snippet

File: ArcGISRuntimeSDKQt_CppSamples/Maps/OpenMobileMap_MapPackage/OpenMobileMap_MapPackage.h [hunk 3]
```diff
@@ -39,9 +42,16 @@ class OpenMobileMap_MapPackage : public QQuickItem
   void componentComplete() override;
   static void init();
 
+private slots:
+  void packageLoaded(const Esri::ArcGISRuntime::Error& e);
```
Comment (lsmallwood) [inline ArcGISRuntimeSDKQt_CppSamples/Maps/OpenMobileMap_MapPackage/OpenMobileMap_MapPackage.h]: optional, but I find it useful to name slots as starting with "on" or "handle" - so `handlePackageLoaded`

File: ArcGISRuntimeSDKQt_QMLSamples/Maps/OpenMobileMap_MapPackage/OpenMobileMap_MapPackage.qml [hunk 4]
```diff
@@ -24,30 +24,74 @@ Rectangle {
...
+            if (mmpk.maps.length < 1) {
+                return;
+            }
+
+            // set the map view's map to the first map in the mobile map package
+            mapView.map = mmpk.maps[0];
+        }
+
+        onErrorChanged: {
+            console.log("Mobile Scene Package Error: %1 %2".arg(error.message).arg(error.additionalMessage));
```
Comment (lsmallwood) [inline ArcGISRuntimeSDKQt_QMLSamples/Maps/OpenMobileMap_MapPackage/OpenMobileMap_MapPackage.qml]: "Mobile Scene Package" => "Mobile Map Package"

File: ArcGISRuntimeSDKQt_CppSamples/Maps/OpenMobileMap_MapPackage/OpenMobileMap_MapPackage.cpp (hunk 1 above)
Comment (mfeigl) [reply ArcGISRuntimeSDKQt_CppSamples/Maps/OpenMobileMap_MapPackage/OpenMobileMap_MapPackage.cpp]: Done.

File: ArcGISRuntimeSDKQt_CppSamples/Maps/OpenMobileMap_MapPackage/OpenMobileMap_MapPackage.h (hunk 3 above)
Comment (mfeigl) [reply ArcGISRuntimeSDKQt_CppSamples/Maps/OpenMobileMap_MapPackage/OpenMobileMap_MapPackage.h]: Very well. This function is now removed, but I will keep it in mind for future reference.

File: ArcGISRuntimeSDKQt_CppSamples/Maps/OpenMobileMap_MapPackage/OpenMobileMap_MapPackage.cpp (hunk 2 above)
Comment (mfeigl) [reply ArcGISRuntimeSDKQt_CppSamples/Maps/OpenMobileMap_MapPackage/OpenMobileMap_MapPackage.cpp]: Done.

**Comment by mfeigl** [comment]
@ldanzinger @lsmallwood Pull request updated with the implemented suggestions.

**Comment by ldanzinger** [comment]
@mfeigl looks good to merge

**Pull request merged**